Custom2DPose new_init_pose
---
Custom2DPose old_final_pose

